### ISSUE TITLE
No longer maintained? Happy to jump in 👋 

### ISSUE THREAD

**Issue by jamilabreu** [liveforeverx/dlex#118]
Hi @liveforeverx! I've been jumping in with Dlex, and noticed you're not as active anymore.

If you could give me a quick overview of how everything works (Confused about Adapters, Protocols, etc), I'm happy to take over. Lmk 😃

**Comment by liveforeverx** [comment]
Hi, @jamilabreu !

It is kind of true for now.

It has multiple reasons from my personal (multiple changes in personal life within short period of time) till changes in Dgraph, on which I'm not sure yet, how to react. I know, that `dlex` needs GraphQL support. But I saw, that DQL was always more powerful and they now starts to add features to GraphQL, which are not supported in DQL. 

This means, that it isn't possible to use only one interface (the most powerful, which was DQL) for DGraph. And because of missing features fro years (sorted indexes, unique indexes which should be added to GraphQL, possibility to have separate test/dev databases), I'm heavily thinking to not use Dgraph in my current personal project. This project was always a driver for Dlex development. Because of this, I'm not sure, if I still will have a good driver for further Dlex development.

All, in all, I'm happy to have new maintainers in place. 
You can mail me and we can have hangouts, telegram, skype or any other possibility to discuss everything in person, dive in a code and discuss further changes. What do you think?

**Comment by liveforeverx** [comment]
Hi, @jamilabreu !

Sorry for late answer.

I'm kind of waiting, what the new direction with GraphQL it will take, so I'm not sure where DGraph goes and where should a driver go at the moment.

But I'm happy to have new maintainers.

You can mail me and we can have hangouts, telegram, skype or any other possibility to discuss everything in person, dive in a code and discuss further changes. What do you think?

**Comment by martinthenth** [comment]
Hi @liveforeverx and @jamilabreu,

We could open an RFC issue, to gather changes to be made for a version 1 release. Dlex does a lot of things right, is up to date on the current version of Dgraph (to my knowledge), and can be used in production (I have).

For example:

1. I believe much of the internal API can be simplified to just `mutate()` and `all()` / `query()`, because `set()` and `mutate()` do a lot of the same things.
2. `set` and `delete` should be possible in a single query as per Dgraph docs (which we can do with pattern matches). 
3. The typespecs have to be updated. 
4. And I believe Ecto and Dlex can be separated, because Ecto can do the heavy lifting of changesets, and Dlex only needs to do Dlex.set(map) where `map` is a result of the Ecto.Changeset `changes` field.

**Comment by jamilabreu** [comment]
@liveforeverx I'd love a quick tour over zoom or hangouts. I'm free next week starting on Tuesday, any day after 5pm EST. Let me know what works for you!

@martinthenth - you're welcome to join too if you'd like! Happy to see there's interest in keeping Dlex up-to-date 😃 An RFC issue sounds like a good start - want to work on it together?

**Comment by liveforeverx** [comment]
@martinthenth Yes, it is up to date, usually I test against every new version.

1. The `set` API should be deprecated. I agree, this was still there on backwards-compatibility reason.
2. It is already possible (I try to cover all functionality, except for GraphQL atm): https://github.com/liveforeverx/dlex/blob/master/test/dlex_test.exs#L285-L288 
3. Yes, that's true
4. Completely agree, for version 1.0 - it should. For future developing, I think, it is better to have one place at the moment. For library users of dlex it shouldn't be a problem, as it optionally depends on `ecto`.

**Comment by martinthenth** [comment]
@jamilabreu Yes, I can help. :)

@liveforeverx Great! I'm glad the base-API of Dgraph is stable through the different versions. 2: I hadn't seen that, my bad.

**Comment by jamilabreu** [comment]
@liveforeverx are you free to chat this week? Here's a calendly link with my available times, looking forward to it!

https://calendly.com/abreu-jamil/30

Github would be better with private messages :)

**Comment by benwoodward** [comment]
Can someone explain the difference between this repo and the one here—https://github.com/pixelmonster-ou/dlex?

**Comment by jamilabreu** [comment]
@benwoodward Hi - I'm not sure they're related, looks like a fork?

**Comment by benwoodward** [comment]
I should have updated here, the author replied with this: 

> Hi @benwoodward,
> 
> The one you linked is the real one. This one just has custom changes I did for https://github.com/pixelmonster-ou/metagraph such as reverse relations, translations etc.
> 
> I would suggest you go with @liveforeverx's repo as it's more clean and more up2date.

**Comment by jnylen** [comment]
I'm fine merging my fork into this one if @liveforeverx wants to.

Basically the fork was made so I could add quite custom stuff to model databases such as reverse relationships and mainly translations as this couldn't be handled in a good way that I expected.

@liveforeverx if you are fine with merging it, do you want me to start splitting up the changes into multiple PRs?

**Comment by geoyws** [comment]
Really hoping this project will pick up again!
I'm learning Elixir and want to use it with Dgraph for a small project 😄

**Comment by m0dnar** [comment]
👋🏼 I've just pushed an updated fork of Dlex with:

- Updated dependencies
- Updated api.ex and api.proto
- Updated fields 
- Deprecated `set` in favor of `mutate`
- Support for redacted fields
- Updated README with Phoenix instructions

@martinthenth would love some help separating Ecto from Dlex 💯

@jnylen happy to accept your work on translations / reverse relationships

https://github.com/m0dnar/dlex

Excited to get this project going again!

cc: @geoyws @benwoodward @liveforeverx

**Comment by benwoodward** [comment]
> 👋🏼 I've just pushed an updated fork of Dlex with:

Fantastic! Are you gonna create a PR or maintain a separate fork?

**Comment by m0dnar** [comment]
> Fantastic! Are you gonna create a PR or maintain a separate fork?

Separate for now, but I'm not against merging down the road!

**Comment by goravbhootra** [comment]
I had raised two PRs which are still not merged. In case they are useful, I will be happy to rebase and update, as needed. Keen on contributing going forward as well.

Cheers!

**Comment by m0dnar** [comment]
@goravbhootra I actually added the changes you suggested in both of your PRs in my fork -- very helpful and appreciated :D Send any others over!

**Comment by tjtrabue** [comment]
Is this project still maintained? It's still by far the best dgraph client for Elixir, and it'd be a shame for it to fall away.
